modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/DynamicInlineScroll.es.js

```diff
@@ -180,12 +190,23 @@ class DynamicInlineScroll extends PortletBase {
 		}
 
 		if (
+			targetNode.nodeName === 'UL' &&
 			cur <= pages &&
 			pageIndex < pageIndexMax &&
-			target.getAttribute('scrollTop') >=
-				target.getAttribute('scrollHeight') - 300
+			targetNode.scrollTop >= targetNode.scrollHeight - 300
 		) {
-			this.addListItem_(target, pageIndex);
+			if (targetNode.nodeName !== target.nodeName) {
+				for (let i = 0; i < 11; i++) {
+					if (pageIndex === pageIndexMax) {
+						break;
+					}
+					this.addListItem_(targetNode, pageIndex);
+					pageIndex++;
+				}
```

Comment:
I tested the PR and, for me, this part never triggers. I suspect this is because navigation to `span` does not change scroll position, as the `span` is invisible.

Do we even need this code? It should be fine if keyboard navigation adds one item, as it always moves by exactly one item.

Reply:
With the code in this PR, the block never triggers for me. The condition
```
targetNode.nodeName !== target.nodeName
```
is always true. More precisely, 
```
targetNode.scrollTop >= targetNode.scrollHeight - 300
```
seems to be false always when above is false.

I am on Chrome, on Ubuntu, without JAWS. But OS and JAWS should not matter here. Navigation is navigation, check depends on HTML and events. There might be some difference in browser behavior, related to scroll window size, but that's very unlikely if we bot use Chrome :thinking:

Reply:
It matters as JAWS is doing a different kind of focus not going on each LI element but on elements inside LI element going even on the span itself.

Here is a video reproducing the error without this fix:
https://user-images.githubusercontent.com/59687465/201986812-f3d4d5d4-e207-4269-b0fe-87135951193f.mp4

As you can see it's not always scrolling to the last when using JAWS navigation with arrow key UP and DOWN, navigating with TAB is working fine as is scrolling as in your video.

This is with the fix:
https://user-images.githubusercontent.com/59687465/201988139-329223de-4bf3-4dfa-8504-37a823c36573.mp4

Not sure why I can't see my js code well on my local, but I added a log after the condition:
```
if (targetNode.nodeName !== target.nodeName) {
				console.log('Condition met');
```

Reply:
I had to change the full solution as testing yesterday I realized that is no way JAWS will scroll correctly through the elements when using UP/DOWN arrows if we are in full screen, these were working when you make the page small, it seems that is scrolling to the full page and not to the list elements.

Here is a video:
https://user-images.githubusercontent.com/59687465/202129795-990350e8-f220-4b70-9024-4ccd58e54581.mp4

What I did was to block adding LI elements if the target is not UL and add a new key to tell the users to use TAB key and not arrows keys-

Please review. thnaks!